src/solver.cpp

```diff
@@ -208,6 +238,10 @@ namespace BlitzL1
         add_scaler(A_Delta_x, diff);
       }
 
+      /*
+        criterion based on a fixed point
+        Q: Why multiply by hess ** 2 (cf. L228)
+      */
       if (sum_sq_hess_diff < prox_newton_epsilon && cd_itr + 1 >= MIN_PROX_NEWTON_CD_ITR)
```

Comment:
I can't make sense of this stopping criterion.
It appears as a weighted fixed point that we compare to the difference between the gradient and its quadratic approximation.